Server/src/models/DatasetModels/DatasetApprovalModel.ts

```diff
@@ -0,0 +1,88 @@
+import { Connection, getConnection } from "typeorm";
+import { Dataset } from "../entities/Dataset";
+import { selectUnapprovedDatasetInfoQuery, Unapproveddatasets } from "../entities/Unapproveddatasets";
+import { IDatasetIDModel } from "../interfaces/DatasetModelInterface";
+import { DatasetCommonModel } from "./DatasetCommonModel";
+
+
+
+export class DatasetApprovalModel {
+    private connection: Connection;
+    private commonModel: DatasetCommonModel
+    constructor() {
+        this.connection = getConnection();
+        this.commonModel = new DatasetCommonModel();
+    }
+
+    private selectDatasetCommentQuery = (id: number) =>
+        this.connection.createQueryBuilder(Dataset, 'dataset')
+            .select('dataset.comments', 'comments')
+            .where('dataset.id = :id', { id: id })
+            .getRawOne();
+
+    async selectUserFlaggedDatasets(uploaderId: number): Promise<IDatasetIDModel[]> {
+        let userFlaggedDatasets = await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_Datasets')
+            .select('unapproved_datasets.datasetId', 'dataset_id')
+            .innerJoin(Dataset, 'dataset.id = unapproved_Datasets.datasetId')
+            .where('unapproved_datasets.isFlagged = 1')
+            .andWhere('dataset.uploaderId = :uploaderId', { uploaderId: uploaderId })
+            .getRawMany();
+        return userFlaggedDatasets
+    }
+
+    async getUnapprovedDatasets(): Promise<IDatasetIDModel[]> {
+        let allUnapprovedDatasets = await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_datasets')
+            .select('unapproved_datasets.datasetId', 'dataset_id')
+            .getRawMany();
+        return allUnapprovedDatasets
+    }
+
+    async selectAllFlaggedDatasets(): Promise<IDatasetIDModel[]> {
+        let allFlaggedDatsets = await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_datasets')
+            .select('unapproved_datasets.datasetId', 'dataset_id')
+            .where('unapproved_datasets.isFlagged = 1')
+            .getRawMany();
+        return allFlaggedDatsets
+    }
+
+    async updateApprovedStatus(datasetId: number) {

```

Comment:
a bit vague, update sounds like can be either set to 1 or 0 but this only approves. can we change it to make sure people know this approves a dataset?

Reply:
agreed

Reply:
Renaming this to: approveSingleDatasetQuery